frontend/src/pages/EditRoom/index.tsx

```diff
@@ -18,38 +20,55 @@ const EditRoomPage: FunctionComponent<EditRoomPageProps> = ({
 }) => {
   const history = useHistory();
   const { rooms, createRoom, updateRoom } = useRooms();
+  const { speakers, updateSpeaker, deleteSpeaker } = useSpeakers();
   const currentRoom = rooms?.find(room => room.id === roomId);
+  const currentRoomSpeakers = speakers?.filter(speaker => speaker.room.id === currentRoom?.id);
 
   const [saving, setSaving] = useState(false);
   const [showSaveSuccess, setShowSaveSuccess] = useState(false);
   const [saveErrors, setSaveErrors] = useState<string[]>();
 
-  const save = (values: { name: string, nodes: string[] }) => {
+  const save = async (values: { name: string, speakers: number[] }) => {
     setSaving(true);
-    if (currentRoom) {
-      updateRoom({ id: currentRoom.id, name: values.name }).then(() => {
-        setSaving(false);
-        setSaveErrors(undefined);
-        setShowSaveSuccess(true);
-        setTimeout(() => setShowSaveSuccess(false), 3000);
-      }).catch((ack: Acknowledgment) => {
-        setSaving(false);
-        setSaveErrors(ack.errors);
-      })
-    } else {
-      createRoom({ name: values.name }).then(ack => {
-        setSaving(false);
-        setSaveErrors(undefined);
-        history.push(`/rooms/${ack.createdId}/edit`);
-        setShowSaveSuccess(true);
-        setTimeout(() => setShowSaveSuccess(false), 3000);
-      }).catch((ack: Acknowledgment) => {
-        setSaving(false);
-        setSaveErrors(ack.errors)
-      });
+    let ack;
+    try {
+      if (currentRoom) {
+        ack = await updateRoom({ id: currentRoom.id, name: values.name })
+      } else {
+        ack = await createRoom({ name: values.name });
+      }
+      const currentRoomId = currentRoom ? currentRoom.id : ack.createdId;
+
+      if (speakers && currentRoomId) {
+        await Promise.all(speakers.map(speaker => {
+          if (speaker.room.id !== currentRoomId && values.speakers.includes(speaker.id)) {
+            return updateSpeaker({ ...speaker, room: { id: currentRoomId } });
+          } else if (speaker.room.id === currentRoomId) {
```

Comment:
When I had speakers assigned to a room and simply clicked the save button (e.g. only change the name but nothing related to the speakers), all speakers got removed. I think this is the line that caused that bug.